lab_4_retrieval_w_clustering/main.py

```diff
@@ -113,6 +163,26 @@ def put_corpus(self, corpus: Corpus) -> None:
                 or if input arguments are empty,
                 or if methods used return None.
         """
+        if (not corpus or not isinstance(corpus, list)
+                or not all(isinstance(element, str) for element in corpus)):
+            raise ValueError
+
+        self._vectorizer.build()
+        self.__documents = corpus
+
+        list_of_tok_paragraphs = []
+        for doc_aka_str in self.__documents:
+            doc_aka_list = Tokenizer.tokenize(self._tokenizer, doc_aka_str)
```

Comment:
really `Tokenizer.tokenize()`? You already have tokenizer in attribute

Reply:
done